### PR TITLE
[Composite Products] Add Component Settings view and view model

### PR DESCRIPTION
**Pull request by rachelmcr**
Part of #8956

## Description

This adds a new Component Settings view for displaying read-only component settings. It displays the provided settings:

* If the image is `nil`, it is hidden (like the read-only product details screen).
* If the description is empty, it is hidden (like the read-only product details screen).
* If the options list is empty, it shows a "No options selected" placeholder.

## Testing instructions

This view is not yet used in the app, so you can check the SwiftUI previews or screenshots below to see how it looks.

## Screenshots

With images and description|Without images and description|Without options
-|-|-
Screenshot 2023-03-24 at 1 56 32 PM|Screenshot 2023-03-24 at 3 28 42 PM|Screenshot 2023-03-24 at 5 30 14 PM

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/Composite Product Components/ComponentSettings.swift [hunk 1]
```diff
@@ -0,0 +1,161 @@
+import SwiftUI
+import Kingfisher
+
+/// Renders the settings for a component in a composite product
+///
+struct ComponentSettings: View {
+
+    /// View model that directs the view content.
+    ///
+    let viewModel: ComponentSettingsViewModel
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Products/Edit Product/Composite Product Components/ComponentSettings.swift:10]: Be aware of this ownership when the view model starts to get more dynamic

**Review by Ecarrion — APPROVED** [review]
Looks good.

Not a blocker, but the backgrounds do not go edge-to-edge.
Screenshot 2023-03-24 at 1 56 04 PM

**Comment by oguzkocer** [comment]
👋 I am working on the code freeze for `13.0`, so I am bumping the milestone for this PR to `13.1`. Please feel free to revert the milestone change and merge this PR to `release/13.0` if this is urgent and give me a heads up so I can generate a new beta.

**Comment by Ecarrion** [comment]
> 👋 I am working on the code freeze for `13.0`, so I am bumping the milestone for this PR to `13.1`. Please feel free to revert the milestone change and merge this PR to `release/13.0` if this is urgent and give me a heads up so I can generate a new beta.

Not urgent as far as I'm aware. Thanks for the heads up!

File: WooCommerce/Classes/ViewRelated/Products/Edit Product/Composite Product Components/ComponentSettings.swift (hunk 1 above)
Comment (rachelmcr) [reply WooCommerce/Classes/ViewRelated/Products/Edit Product/Composite Product Components/ComponentSettings.swift:10]: Thanks for the reminder! I've made a note to check on this.

**Comment by rachelmcr** [comment]
> Not a blocker, but the backgrounds do not go edge-to-edge.

🤦‍♀️ I meant to check that and somehow missed it. Fixed in 08fd0f1.

Screenshot 2023-04-03 at 2 06 28 PM
Screenshot 2023-04-03 at 2 06 37 PM

**Pull request merged**